contrib/verifybinaries/verify.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Script for verifying Bitoin Core release binaries
+
+This script attempts to download the signature file SHA256SUMS.asc from
+bitcoincore.org and bitcoin.org and compares them.
+It first checks if the signature passes, and then downloads the files
+specified in the file, and checks if the hashes of these files match those
+that are specified in the signature file.
+The script returns 0 if everything passes the checks. It returns 1 if either
+the signature check or the hash check doesn't pass. If an error occurs the
+return value is >= 2.
+"""
+import os
+import subprocess
+import sys
+from textwrap import indent
+
+WORKINGDIR = "/tmp/bitcoin_verify_binaries"
+HASHFILE = "hashes.tmp"
+HOST1 = "https://bitcoincore.org"
+HOST2 = "https://bitcoin.org"
+VERSIONPREFIX = "bitcoin-core-"
+SIGNATUREFILENAME = "SHA256SUMS.asc"
+
+
+def parse_version_string(version_str):
+    if version_str.startswith(VERSIONPREFIX):  # remove version prefix
+        version_str = version_str[len(VERSIONPREFIX):]
+
+    parts = version_str.split('-')
+    version_base = parts[0]
+    version_rc = ""
+    version_os = ""
+    if len(parts) == 2:  # "<version>-rcN" or "version-platform"
+        if "rc" in parts[1]:
+            version_rc = parts[1]
+        else:
+            version_os = parts[1]
+    elif len(parts) == 3:  # "<version>-rcN-platform"
+        version_rc = parts[1]
+        version_os = parts[2]
+
+    return version_base, version_rc, version_os
+
+
+def download_with_wget(remote_file, local_file=None):
+    if local_file:
+        wget_args = ['wget', '-N', '-O', local_file, remote_file]
+    else:
+        wget_args = ['wget', '-N', remote_file]
+
+    result = subprocess.run(wget_args,
+                            stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    return result.returncode == 0, result.stdout.decode().rstrip()
+
+
+def verify_with_gpg(signature_filename, output_filename):
+    result = subprocess.run(['gpg', '--yes', '--decrypt', '--output',
+                             output_filename, signature_filename],
+                             stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    return result.returncode, result.stdout.decode().rstrip()
+
+
+def apply_filter_on_file(filename, filter_):
+    subprocess.run(f'grep {filter_} {filename} > {filename}-tmp', shell=True)
+    if os.path.getsize(f"{filename}-tmp") == 0:
+        return False
+    subprocess.run(f"mv {filename}-tmp {filename}", shell=True)
+    return True
+
+
+def verify_with_sha256sum(files_to_verify, hashfile):
```

Comment:
This is a missed oppertunity; please use python's internal hashing instead of calling out to an utility here.

Reply:
Good point! When I thought about using internal hashing I was initially put off about potentially loading dozens of MBs into memory (I guess `sha256sum` does this in a smart way by loading chunk after chunk, instead of the whole file at once), but OTOH this should not be a problem at all and the advantage of getting rid of an external tool outweighs this by far.

Reply:
I'm late to the party, but you can do stream hashing in python:

```python
import hashlib as hash

CHUNKSIZE = 65536

sha = hash.sha256()
with open('/path/to/foo.bar', 'rb') as fh:
    buff = fh.read(CHUNKSIZE)
    while len(buff) > 0:
        sha.update(buff)
        buff = fh.read(CHUNKSIZE)
hash = sha.hexdigest()
```

Reply:
@shesek: Thanks for posting! I didn't doubt that stream hashing is possible at all in python, I just was put off by the idea of partly reinventing the wheel of `sha256` here and investing more than 2-3 LOC for getting the hash of a file. 

> I'm late to the party, ...

Well, luckily in open source development there is no "late to the party", you can always open a follow-up PR :)